lms/djangoapps/verify_student/models.py

```diff
@@ -1189,3 +1190,28 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+
+class VerificationAttempt(TimeStampedModel):
+    """
+    The model represents impelementation-agnostic information about identity verification (IDV) attempts.
+
+    Plugins that implement forms of IDV can store information about IDV attempts in this model for use across
+    the platform.
+    """
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
+    name = models.CharField(blank=True, max_length=255)
```

Comment:
Is a blank name really ever okay?

Reply:
It depends. Based on the current Software Secure flow, it is possible but is difficult and likely occurs seldomly. You'd have to really try to get around providing a name, likely by hitting the legacy IDV flow somehow.

The [SoftwareSecurePhotoVerification instance can be created without a name](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/verify_student/views.py#L985-L987) if the [full name isn't provided during IDV](https://github.com/openedx/edx-platform/blob/688242abd9937559e655166e6c7fc6339f4e1acc/lms/static/js/verify_student/models/verification_model.js#L49-L58) - this is the legacy view but is also true of the [MFE based flow](https://github.com/openedx/frontend-app-account/blob/c5c05c86f73c15a520b681ca09a0e969d04b3772/src/id-verification/panels/SummaryPanel.jsx#L31). The name is added in a subsequent [mark_ready step](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/verify_student/models.py#L403-L404), but `profile_name` is an optional field of the `UserProfile` model, so it can be empty.

I don't know the full history of the choice for `blank=True`, but my guess is that before we added the Name Affirmation feature, we only cared that the user was ID verified, and the name was secondary to that fact and was only used optionally for the purposes of a name change - it wasn't necessary to supply a name to become IDV verified. Now, you need to supply a name before IDV because of the Name Affirmation feature; although, there are ways around it.

Also, an `SSOVerification` can be [created without a name](https://github.com/openedx/edx-platform/blob/688242abd9937559e655166e6c7fc6339f4e1acc/common/djangoapps/third_party_auth/pipeline.py#L951) because `profile_name` is an optional field. If we adopt the position that IDV is used to verify names and identities, then we could only create `SSOVerifications` only when the learner has a name to verify and that has been [synced from the third party provider](https://github.com/openedx/edx-platform/blob/688242abd9937559e655166e6c7fc6339f4e1acc/common/djangoapps/third_party_auth/pipeline.py#L869).

I don't think `ManualVerification` or our Persona integration will make sense without a name, but I tried to mimic the schema of the `IDVerificationAttempt` model here. I'm comfortable making it required based on those two verification methods, but I'm unsure about `SSOVerification`, and I don't know if additional implementations of IDV may require an IDV attempt without name.

If that's something we'd like to enforce, then I can make it `required`, but it's going to require some help from product re: `SSOVerifications` and will complicate process of deprecating the Software Secure integration. When deprecating the `SoftwareSecurePhotoVerification` model and migrating the data into `VerificationAttempt`, we'll have to drop any instances without a `name` value.

Do you have any thoughts?

Reply:
I asked mostly just to check that it wasn't a simple oversight. If this were something being built from scratch, I might push to disallow blank entries here. But I'm not looking to do anything that would complicate the migration process, so `blank=True` it is.